tapdb/sqlc/queries/transfers.sql

```diff
@@ -42,20 +42,25 @@ SET proof_delivery_complete = @delivery_complete
 WHERE output_id = (SELECT output_id FROM target);
 
 -- name: QueryAssetTransfers :many
-SELECT
-    id, height_hint, txns.txid, transfer_time_unix
+SELECT DISTINCT
+    id, height_hint, txns.txid, txns.block_hash AS anchor_tx_block_hash,
+    transfer_time_unix
 FROM asset_transfers transfers
+JOIN asset_transfer_outputs outputs
+    ON outputs.transfer_id = transfers.id
 JOIN chain_txns txns
-    ON transfers.anchor_txn_id = txns.txn_id
--- We'll use this clause to filter out for only transfers that are
--- unconfirmed. But only if the unconf_only field is set.
-WHERE (@unconf_only = false OR @unconf_only IS NULL OR
-    (CASE WHEN txns.block_hash IS NULL THEN true ELSE false END) = @unconf_only)
+    ON txns.txn_id = transfers.anchor_txn_id
+WHERE
+    -- Optionally filter on a given anchor_tx_hash.
+    (txns.txid = sqlc.narg('anchor_tx_hash')
+        OR sqlc.narg('anchor_tx_hash') IS NULL)
 
--- Here we have another optional query clause to select a given transfer
--- based on the anchor_tx_hash, but only if it's specified.
-AND (txns.txid = sqlc.narg('anchor_tx_hash') OR
-    sqlc.narg('anchor_tx_hash') IS NULL)
+    -- Filter for pending transfers only if requested.
+    AND (
+        (@pending_transfers_only = true
+             AND (txns.block_hash IS NULL OR outputs.proof_delivery_complete = false))
```

Comment:
Can't use `COALESCE` in postgres with two different types. I propose `txns.block_hash IS NULL OR txns.block_hash = X'00'` as an alternative.

Reply:
OK. `txns.block_hash = X'00'` doesn't work in Postgres. And `txns.block_hash = '\x00'::bytea` doesn't work in sqlite. Any ideas on this? Are we trying to stop identify when `txns.block_hash` is an all zeros `[32]byte`? We should ensure that it isn't set instead?

Reply:
Agree, since it's nullable I think we shouldn't need to be storing a blob of `0`?

Reply:
Sorry, I now see that my whole comment was super wrong... Even though I clearly even wrote `block_hash`, in my mind I was reading `block_height` the whole time... And because I recently did something in that area I knew that we basically treated `NULL` and `0` as the same there. But of course this doesn't make any sense for the **hash**.
Sorry, I'm all over the place lately, can't concentrate properly with the heat and lack of sleep...

Reply:
No problem! :+1: Null check only it is.